lib/js/src/manager/screen/utils/SoftButtonObject.js

```diff
@@ -363,10 +363,23 @@ class SoftButtonObject {
                 if (states[index].getSystemAction() !== otherStates[index].getSystemAction()) {
                     return false;
                 }
-                if (states[index].getSoftButton() === null && otherStates[index].getSoftButton() === null) {
-                    continue;
+                if (states[index].getSoftButton().getType() !== otherStates[index].getSoftButton().getType()) {
+                    return false;
+                }
+                if (states[index].getSoftButton().getText() !== otherStates[index].getSoftButton().getText()) {
+                    return false;
+                }
+                if (states[index].getSoftButton().getImage() !== otherStates[index].getSoftButton().getImage()) {
+                    return false;
+                }
+                if (states[index].getSoftButton().getSoftButtonID() !== otherStates[index].getSoftButton().getSoftButtonID()) {
+                    return false;
+                }
+                // The SdlFile equals() method checks most but not all of the SdlArtwork fields
+                if (states[index].getArtwork().equals(otherStates[index].getArtwork())) {
+                    return false;
                 }
-                if (states[index].getSoftButton().getParameters() !== otherStates[index].getSoftButton().getParameters()) {
+                if (states[index].getArtwork().isTemplateImage() !== otherStates[index].getArtwork().isTemplateImage()) {
                     return false;
                 }
             }
```

Comment:
I'm still getting issues with the equals method here. See the following for some code snippets that should return true:

```
        const m1 = new SDL.manager.screen.utils.SoftButtonObject('hello', new SDL.manager.screen.utils.SoftButtonState('hi', 'ok', new SDL.manager.file.filetypes.SdlArtwork().setType(SDL.rpc.enums.FileType.GRAPHIC_JPEG)), 'hi')
        console.log(m1.clone().equals(m1));
```

```
const m1 = new SDL.manager.screen.utils.SoftButtonObject('hello', new SDL.manager.screen.utils.SoftButtonState('hi', 'ok'), 'hi')
        const m2 = new SDL.manager.screen.utils.SoftButtonObject('hello', new SDL.manager.screen.utils.SoftButtonState('hi', 'ok'), 'hi')
        console.log(m1.equals(m2));
```

```
const m1 = new SDL.manager.screen.utils.SoftButtonObject('hello', new SDL.manager.screen.utils.SoftButtonState('hi', 'ok', new SDL.manager.file.filetypes.SdlArtwork().setType(SDL.rpc.enums.FileType.GRAPHIC_JPEG)), 'hi')
        const m2 = new SDL.manager.screen.utils.SoftButtonObject('hello', new SDL.manager.screen.utils.SoftButtonState('hi', 'ok', new SDL.manager.file.filetypes.SdlArtwork().setType(SDL.rpc.enums.FileType.GRAPHIC_JPEG)), 'hi')
        console.log(m1.equals(m2));
```

Reply:
The check using the SdlFile equals method needed to be inverted. These code snippets pass with the most recent commit.